omniscidb/QueryEngine/InputMetadata.cpp

```diff
@@ -54,23 +52,30 @@ void InputTableInfoCache::clear() {
 
 namespace {
 
-bool uses_int_meta(const hdk::ir::Type* col_type) {
-  return col_type->isInteger() || col_type->isDecimal() || col_type->isDateTime() ||
-         col_type->isBoolean() || col_type->isExtDictionary();
-}
-
 TableFragmentsInfo synthesize_table_info(hdk::ResultSetTableTokenPtr token) {
   std::vector<FragmentInfo> result;
   bool non_empty = false;
-  for (int frag_id = 0; frag_id < token->resultSetCount(); ++frag_id) {
+  for (int frag_id = 0; frag_id < (int)token->resultSetCount(); ++frag_id) {
```

Comment:
I have always preferred `static_cast`, but I wouldn't change it if that's the only change you need to make in this PR.

Reply:
Fixed